app/javascript/authentication/SignInForm.tsx

```diff
@@ -10,40 +10,106 @@ import {
   AlertBox,
   SiteAlert,
   LinkButton,
+  debounce,
 } from "@bloom-housing/ui-components"
-import { useForm } from "react-hook-form"
+import { Dialog, Link, Heading, Alert } from "@bloom-housing/ui-seeds"
+import { FieldValues, SubmitHandler, useForm } from "react-hook-form"
 
 import { getMyAccountPath } from "../util/routeUtil"
-import UserContext from "./context/UserContext"
 import EmailFieldset from "../pages/account/components/EmailFieldset"
 import PasswordFieldset from "../pages/account/components/PasswordFieldset"
-import { Link, Heading } from "@bloom-housing/ui-seeds"
 import "../pages/account/styles/account.scss"
+import { AxiosError } from "axios"
+import { confirmEmail } from "../api/authApiService"
+import UserContext from "./context/UserContext"
 
-const SignInForm = () => {
-  const { signIn } = useContext(UserContext)
-  /* Form Handler */
-  // TODO(DAH-1575): Upgrade React-Hook-Form. Note: When you update to Version 7 of react-hook-form, "errors" becomes: "formState: { errors }""
-  // This is causing a linting issue with unbound-method, see open issue as of 10/21/2020:
-  // https://github.com/react-hook-form/react-hook-form/issues/2887
-  // eslint-disable-next-line @typescript-eslint/unbound-method
-  const { register, handleSubmit, errors, watch } = useForm()
-  const [requestError, setRequestError] = useState<string>()
-
-  const onSubmit = (data: { email: string; password: string }) => {
-    const { email, password } = data
+const AccountAlreadyConfirmedModal = ({
```

Comment:
**question (non-blocking):** Is it possible to pull these Modals into their own file?

And is there any opportunity for having all of these modals use a new "base" Modal component?

`SignInForm.tsx` is getting pretty long. I am okay with handling this cleanup next sprint, though.

Reply:
Yes!

Reply:
I'm going to wait until the base pr is done before making any more changes

Reply:
@cade-exygy I moved the modals to their own file!!

I tried to make a base modal component, but it actually made it way more complicated - I think the two modals are two dissimilar

Reply:
Awesome! and that makes sense, thanks for taking a look!